fgpyo/sam/__init__.py

```diff
@@ -1266,8 +1190,220 @@ class SamOrder(enum.Enum):
 
 
 @dataclass(frozen=True)
-class SecondaryAlignment:
-    """A secondary alignment as encoded in one part of the `XA` or `XB` tag value on a SAM record.
+class AuxAlignment(ABC, DataclassInstance):
+    """An alignment parsed from the data stored in the SAM tags of a SAM record.
+
+    Args:
+        reference_name: The reference sequence name.
+        reference_start: The 0-based start position of the alignment.
+        is_forward: If the alignment is in the forward orientation or not.
+        cigar: The Cigar sequence representing the alignment.
+        mapq: The aligner-reported probability of an incorrect mapping, if available.
+        edit_distance: The number of mismatches between the query and the target, if available.
+        alignment_score: The aligner-reported alignment score, if available.
+
+    Raises:
+        ValueError: If reference_start is set to a value less than zero.
+        ValueError: If mapq is set to a value less than zero.
+        ValueError: If edit_distance is set to a value less than zero.
+        ValueError: If alignment_score is set to a value less than zero.
+    """
+
+    sam_tags: ClassVar[Collection[str]]
```

Comment:
class constants should be upper snake case ([link](https://google.github.io/styleguide/pyguide.html#3164-guidelines-derived-from-guidos-recommendations))
```suggestion
    SAM_TAGS: ClassVar[Collection[str]]
```